DSCResources/MSFT_xADDomain/MSFT_xADDomain.psm1

```diff
@@ -3,25 +3,37 @@ data localizedData
-        QueryDomainWithCredential      = Computer is a workgroup member; querying for domain '{0}' using supplied credential ...
-        DomainFound                    = Active Directory domain '{0}' found.
-        DomainNotFound                 = Active Directory domain '{0}' cannot be found.
-        CreatingChildDomain            = Creating domain '{0}' as a child of domain '{1}' ...
-        CreatedChildDomain             = Child domain '{0}' created.
-        CreatingForest                 = Creating AD forest '{0}' ...
-        CreatedForest                  = AD forest '{0}' created.
-        ResourcePropertyValueIncorrect = Property '{0}' value is incorrect; expected '{1}', actual '{2}'.
-        ResourceInDesiredState         = Resource '{0}' is in the desired state.
-        ResourceNotInDesiredState      = Resource '{0}' is NOT in the desired state.
+        RoleNotFoundError                    = Please ensure that the PowerShell module for role '{0}' is installed.
+        InvalidDomainError                   = Computer is a member of the wrong domain?!
+        ExistingDomainMemberError            = Computer is already a domain member. Cannot create a new '{0}' domain?
+        InvalidCredentialError               = Domain '{0}' is available, but invalid credentials were supplied.
+                                             
+        QueryDomainWithLocalCredential       = Computer is a domain member; querying domain '{0}' using local credential ...
+        QueryDomainWithCredential            = Computer is a workgroup member; querying for domain '{0}' using supplied credential ...
+        DomainFound                          = Active Directory domain '{0}' found.
+        DomainNotFound                       = Active Directory domain '{0}' cannot be found.
+        CreatingChildDomain                  = Creating domain '{0}' as a child of domain '{1}' ...
+        CreatedChildDomain                   = Child domain '{0}' created.
+        CreatingForest                       = Creating AD forest '{0}' ...
+        CreatedForest                        = AD forest '{0}' created.
+        ResourcePropertyValueIncorrect       = Property '{0}' value is incorrect; expected '{1}', actual '{2}'.
+        ResourceInDesiredState               = Resource '{0}' is in the desired state.
+        ResourceNotInDesiredState            = Resource '{0}' is NOT in the desired state.
+        RetryingGetADDomain                  = Attempt {0} of {1} to call Get-ADDomain failed, retrying in {2} seconds.
+    UnhandledError                       = Unhandled error occured, detail here: {0}
+    FaultExceptionAndDomainShouldExist = ServiceModel FaultException detected and domain should exist, performing retry...
+
 '@
 }
 
+function Get-TrackingFilename {
+param(
+    [Parameter(Mandatory)]
+    [String] $DomainName
+    )
+    Join-Path $Env:TEMP ('{0}.xADDomain.completed' -f $DomainName)
```

Comment:
I have left it as temp because if the resource is called again and a domain promotion has already occurred, the resource should not attempt to promote the machine to a domain controller again. The second attempt to promote the domain controller would throw the same error this code is trying to fix.
I have added a comment to the code explaining this and providing the resource cache location if it is needed in the future.